lscache.js

```diff
@@ -152,15 +152,22 @@
 
   function eachKey(fn) {
     var prefixRegExp = new RegExp('^' + CACHE_PREFIX + escapeRegExpSpecialCharacters(cacheBucket) + '(.*)');
-    // Loop in reverse as removing items will change indices of tail
-    for (var i = localStorage.length-1; i >= 0 ; --i) {
-      var key = localStorage.key(i);
+    // We first identify which keys to process
+    var keysToProcess = [];
+    var key, i;
+    for (i = 0; i < localStorage.length; i++) {
+      key = localStorage.key(i);
       key = key && key.match(prefixRegExp);
       key = key && key[1];
       if (key && key.indexOf(CACHE_SUFFIX) < 0) {
-        fn(key, expirationKey(key));
+        keysToProcess.push(key);
       }
     }
+    // Then we apply the processing function to each key
+    for (i = 0; i < keysToProcess.length; i++) {
+      key = keysToProcess[i];
+      fn(key, expirationKey(key));

```

Comment:
What's the point in an extra variable here? The variable is not passed by reference. You can pass the value directly.

```diff
- key = keysToProcess[i];
- fn(key, expirationKey(key));
+ fn(keysToProcess[i], expirationKey(keysToProcess[i]));
```